File: sds - ML - Python/simple_linear_regression.py
# Simple Linear Regression

# Importing the Libraries

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#importing the dataset
dataset = pd.read_csv('C:/Users/Admin/Desktop/Machine Learning/SDS- ML - A2Z/Machine Learning A-Z/Part 2 - Regression/Section 4 - Simple Linear Regression/Simple_Linear_Regression/Salary_Data.csv')

#creating matrix of features
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,1].values

#Splitting the dataset into Training set and Test set
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=1/3,random_state =0)

#Feature scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""

#Fitting Simple Linear Regression to the Training set
from sklearn.linear_model import LinearRegression 
regressor = LinearRegression()
regressor.fit(X_train,y_train)

#Predicting the Test set results
y_pred  = regressor.predict(X_test)

#visualising the Training set results
plt.scatter(X_train,y_train,color='red')
plt.plot(X_train,regressor.predict(X_train),color='blue')
plt.title('Salary vs Experience (Training set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

#visualising the Test set results
plt.scatter(X_test,y_test,color='red')
plt.plot(X_train,regressor.predict(X_train),color='blue')
plt.title('Salary vs Experience (Test set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: sds - ML - R/simple_linear_regression.R
#simple linear regression

#importing the dataset
dataset = read.csv('Salary_Data.csv')
# dataset = dataset[,2:3]

#Splitting the dataset into Training set and Test set
#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary,SplitRatio=2/3)
training_set = subset(dataset,split == TRUE)
test_set =subset(dataset,split == FALSE)

#fitting Simple Linear Regression to the training set
regressor = lm(formula= Salary ~ YearsExperience,data=training_set)

#predicting the test set results
y_pred = predict(regressor,newdata = test_set)

#visualising the training set results
#install.packages('ggplot2')
library(ggplot2)
ggplot() + geom_point(aes(x=training_set$YearsExperience,y=training_set$Salary),
                      colour = 'red') + 
          geom_line(aes(x=training_set$YearsExperience,y = predict(regressor,newdata=training_set)),
          colour = 'blue') +
          ggtitle('Salary vs Experience (Training set)') +
          xlab('Years of experience') +
          ylab('Salary')

#visualising the test set results
ggplot() + geom_point(aes(x=test_set$YearsExperience,y=test_set$Salary),
                      colour = 'red') + 
  geom_line(aes(x=training_set$YearsExperience,y = predict(regressor,newdata=training_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (Test set)') +
  xlab('Years of experience') +
  ylab('Salary')
